gnpy/core/network.py

```diff
@@ -122,7 +139,7 @@ def select_edfa(raman_allowed, gain_target, power_target, equipment, uid):
                     -edfa.gain_min,
                 nf=edfa_nf(gain_target, edfa_variety, equipment)) \
                 for edfa_variety, edfa in edfa_dict.items()
-                if (edfa.allowed_for_design and not edfa.raman)]
```

Comment:
Yes this part for rejected edfa.raman was and is still used ('and not edfa.raman' is still a condition for building this list)  because the filtering on suitable amp is not the same for Edfa and Raman amps.
Merging the future raman feature with autodesign will effectively have impacts. I have planned to work on that after july.
The condition on restriction is only here to enable amps from the restriction list and which are not allowed for the autodesign otherwise.
- allowed_for_design restricts the amps to use in general,
- restriction limits the amps to use for roadms only and overrides the allowed_for_design flag
this corresponds to real case where specific amp cards exist for some Roadms but not for others and not for ILA.

Reply:
It looks like I made a mistake interpretting the diff, sorry.